src/wallet/wallet.cpp

```diff
@@ -703,7 +713,7 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         Lock();
         Unlock(strWalletPassphrase);
 
-        // if we are using HD, replace the HD seed with a new one
+        // if we are using HD, create a fresh seed or replace the existing seed with a new one
```

Comment:
I don't see how "create a fresh seed" happens here, because if there's no seed `IsHDEnabled()` returns false.

Reply:
Outdated comment. I've changed it back.

Reply:
Might be useful to add "If no HD seed is set, e.g. because it is a blank wallet, do not add one.", can be done in the same followup as renaming `IsHDEnabled()`.